components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutionResponseProcessor.java

```diff
@@ -45,4 +46,9 @@ ActionExecutionStatus processErrorResponse(Map<String, Object> eventContext,
                                                Event actionEvent,
                                                ActionInvocationErrorResponse errorResponse) throws
             ActionExecutionResponseProcessorException;
+
+    ActionExecutionStatus processFailureResponse(Map<String, Object> eventContext,
+                                               Event actionEvent,
+                                               ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException;
```

Comment:
Given that this is implemented by downstream components this become a change to the api contract.
Let's add a default implementation as not all actions will have failure states.

Reply:
Addressed with commit 0812bd15f91ba1f366b024b26a5aa4e8178d752f